### PR TITLE
Add identity fields update in SampleApp. RD-30694

### PR DESCRIPTION
**Pull request by dhouha123**
Add identity fields update in SampleApp

### CODE REVIEW

**Comment by dhouha123** [comment]
allfields

File: app/src/main/java/com/dimelo/sampleapp/RcConfigurationActivity.java [hunk 1]
```diff
@@ -81,6 +117,41 @@ public void update(View v) {
             dimelo.setUserIdentifier(textInputLayout.getEditText().getText().toString());
         }
 
+        if (userNameInputLayout.getEditText().getText().toString() != null && !userNameInputLayout.getEditText().getText().toString().contains(" ")) {
```
Comment (waelba) [inline app/src/main/java/com/dimelo/sampleapp/RcConfigurationActivity.java]: `toString()` cannot be null so you have to check if `userNameInputLayout != null` && `userNameInputLayout.getEditText() != null`

Comment (dhouha123) [reply app/src/main/java/com/dimelo/sampleapp/RcConfigurationActivity.java]: yes i can check this like that !textInputLayout.getEditText().getText().toString().contains(" ")

**Comment by waelba** [comment]
one minor remark otherwise it works as expected cc @DrissTM

File: app/src/main/java/com/dimelo/sampleapp/RcConfigurationActivity.java (hunk 1 above)
Comment (dhouha123) [reply app/src/main/java/com/dimelo/sampleapp/RcConfigurationActivity.java]: 7267aad6bf0ca71a050cbd6ae37249737f3357e5

File: app/src/main/java/com/dimelo/sampleapp/RcConfig.java [hunk 2]
```diff
@@ -21,6 +21,13 @@
 
 public class RcConfig {
     static final String RC_USER_ID = "rc_user_id";
+    static final String RC_USER_NAME_VAL = "rc_user_name_val";
+    static final String RC_COMPANY_VAL = "rc_company_val";
+    static final String RC_EMAIL_VAL = "rc_email_val";
+    static final String RC_LASTNAME_VAL = "rc_lastname";
+    static final String RC_FIRSTNAME_VAL = "rc_firstname";
+    static final String RC_MOBILE_PHONE_VAL = "rc_mobile_phone";
+    static final String RC_HOME_PHONE_VAL = "rc_home_phone";
```
Comment (DrissTM) [inline app/src/main/java/com/dimelo/sampleapp/RcConfig.java]:
```suggestion
    static final String RC_USER_NAME = "rc_user_name";
    static final String RC_COMPANY = "rc_company";
    static final String RC_EMAIL = "rc_email";
    static final String RC_LASTNAME = "rc_lastname";
    static final String RC_FIRSTNAME = "rc_firstname";
    static final String RC_MOBILE_PHONE = "rc_mobile_phone";
    static final String RC_HOME_PHONE = "rc_home_phone";
```

File: app/src/main/java/com/dimelo/sampleapp/RcConfig.java [hunk 3]
```diff
@@ -38,7 +45,43 @@ static Dimelo setupDimelo(Context context) {
         }
 
         dimelo.setDebug(true);
-        dimelo.setUserName("John Doe");
+
+        String userNameVal = RcConfig.getStringValueFromSharedPreference(context, RC_USER_NAME_VAL);
```
Comment (DrissTM) [inline app/src/main/java/com/dimelo/sampleapp/RcConfig.java]: I'll let you remove the other `Val`

```suggestion
        String userName = RcConfig.getStringValueFromSharedPreference(context, RC_USER_NAME_VAL);
```

File: app/src/main/java/com/dimelo/sampleapp/RcConfigurationActivity.java [hunk 4]
```diff
@@ -31,9 +38,38 @@ public void onCreate(Bundle savedInstanceState) {
         setSupportActionBar(myToolbar);
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
         textInputLayout = findViewById(R.id.user_id);
+        userNameInputLayout = findViewById(R.id.user_name);
+        companyInputLayout = findViewById(R.id.company);
+        emailInputLayout = findViewById(R.id.email);
+        firstnameInputLayout = findViewById(R.id.firstname);
+        lastnameInputLayout = findViewById(R.id.lastname);
+        homePhoneInputLayout = findViewById(R.id.homephone);
+        mobilePhoneInputLayout = findViewById(R.id.mobilephone);
```
Comment (DrissTM) [inline app/src/main/java/com/dimelo/sampleapp/RcConfigurationActivity.java:47]: Please add some clean localized strings for these inputs

**Review by DrissTM — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/dimelo/sampleapp/RcConfigurationActivity.java (hunk 4 above)
Comment (dhouha123) [reply app/src/main/java/com/dimelo/sampleapp/RcConfigurationActivity.java:47]: 544adbfa1a3524981142399bd5f57ad941cfcb64

File: app/src/main/res/values-de/strings.xml [hunk 5]
```diff
@@ -1,3 +1,11 @@
 <resources>
     <string name="rc_sdk_version">SDK version: </string>
+    <string name="user_identifier">User Identifier</string>
+    <string name="username">Username</string>
+    <string name="company">Company</string>
+    <string name="email">Email</string>
+    <string name="firstname">Firstname</string>
+    <string name="lastname">Lastname</string>
+    <string name="home_phone">Home Phone</string>
+    <string name="mobile_phone">Mobile Phone</string>
```
Comment (DrissTM) [inline app/src/main/res/values-de/strings.xml]: these are not translated, we can remove them and it should use the english ones by default, right?

**Review by DrissTM — CHANGES_REQUESTED** [review]

File: app/src/main/res/values-de/strings.xml (hunk 5 above)
Comment (dhouha123) [reply app/src/main/res/values-de/strings.xml]: 2c38e8ca5cf22226ade1ffbf82ea474d44cc2898

File: app/src/main/java/com/dimelo/sampleapp/RcConfigurationActivity.java (hunk 1 above)
Comment (DrissTM) [reply app/src/main/java/com/dimelo/sampleapp/RcConfigurationActivity.java]: @dhouha123 please resolve the discussion when you're done with it

File: app/src/main/res/values-de/strings.xml (hunk 5 above)
Comment (dhouha123) [reply app/src/main/res/values-de/strings.xml]: 6f72596868b6b6b66a401a0ef859e5b02779ac7a

**Review by DrissTM — APPROVED** [review]
LGTM :+1:

**Pull request merged**